src/features/game/Game.tsx

```diff
@@ -46,12 +47,22 @@ const SHOW_MODAL: Record<StateValues, boolean> = {
 };
 
 export const Game: React.FC = () => {
+  const [isOffline, setIsOffline] = useState(false);
```

Comment:
Nice work @Pranav-yadav 

One suggestion would be to move the offline state into the gameMachine. This will ensure that any events (I.e. autosave) don't accidentally get triggered